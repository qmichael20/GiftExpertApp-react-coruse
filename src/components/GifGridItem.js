import React from "react";
import PropTypes from "prop-types";

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__pulse" align="center">
      <img src={url} alt={title}></img>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
