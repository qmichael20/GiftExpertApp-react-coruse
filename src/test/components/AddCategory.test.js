import React from "react";
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";
import "@testing-library/jest-dom";

describe("Pruebas AddCategory", () => {
  const setcategories = jest.fn();
  let wrapper = shallow(<AddCategory setcategories={setcategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setcategories={setcategories} />);
  });

  test("Mostrarse Correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Change en la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola pepe";
    input.simulate("change", {
      target: {
        value,
      },
    });
  });

  test("No debe postear la informacion con submit cuando no hay texto", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setcategories).not.toHaveBeenCalled();
  });

  test("Debe llamar el setcategories y limpiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola Guapo";

    input.simulate("change", {
      target: {
        value,
      },
    });

    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setcategories).toHaveBeenCalledTimes(1);
    expect(setcategories).toHaveBeenCalledWith(expect.any(Function));

    const textValue = wrapper.find("input");

    expect(textValue.prop("value")).toBe("");
  });
});
