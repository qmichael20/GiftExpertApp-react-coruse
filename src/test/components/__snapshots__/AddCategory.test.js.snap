// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas AddCategory Mostrarse Correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <h4 />
  <input
    onChange={[Function]}
    placeholder="Escribe una categoria"
    type="text"
    value=""
  />
</form>
`;
